msbrainpy/quantify/processing.py

```diff
@@ -9,6 +21,128 @@
 from msbrainpy.base import extraListNesting
 
 
+# ------------------------------------------ 2D In Situ Related Functions ----------------------------------------------
+
+def do_gene_series(directory, out_directory, image_name_pattern=r'image_id-\d*.jpeg', find_tissue_mask=False, 
+                   save_pc1=True):
+    if not os.path.exists(out_directory):
+        os.mkdir(out_directory)
+    # find the files
+    files = []
+    image_name_pattern = re.compile(image_name_pattern)
+    for file in sorted(os.listdir(directory)):
+        match = image_name_pattern.match(file)
+        if match is not None:
+            files.append(os.path.join(directory, match[0]))
+    # iterate though the images
+    image_collection = io.imread_collection(files)
+    for i in range(len(image_collection)):
+        white_balanced = white_balancing(image_collection[i])
+        if find_tissue_mask:
+            tissue_mask, greyscale = tissue_segmentation(white_balanced)
+            mask_name = str(i) + '_tissue_mask.tif'
+            mask_path = os.path.join(out_directory, mask_name)
+            tissue_name = str(i) + '_slice.tif'
+            tissue_path = os.path.join(out_directory, tissue_name)
+            io.imsave(mask_path, tissue_mask)
+            io.imsave(tissue_path, greyscale)
+        segmented, pc1_image = find_expressing_pixels(white_balanced)
+        segmented_name = str(i) + '_segmentation.tif'
+        segmented_path = os.path.join(out_directory, segmented_name)
+        io.imsave(segmented_path, segmented)
+        if save_pc1:
+            pc1_image = util.img_as_ubyte(pc1_image)
+            pc1_image_name = str(i) + '_pc1_greyscale.tif'
+            pc1_image_path = os.path.join(out_directory, pc1_image_name)
+            io.imsave(pc1_image_path, pc1_image)
+
+
+def find_expressing_pixels(image_rgb, binomial_kernel=binomial_kernel_31):
+    highest_var = rgb_PCA(image_rgb.copy())
+    median = filters.median(highest_var)
+    edge_enhanced = filters.sobel(median)
+    binary = edge_enhanced >= threshold_otsu(edge_enhanced)
+    binary = morphology.binary_closing(binary, morphology.disk(3))
+    binary = morphology.remove_small_holes(binary)
+    return binary, highest_var
+
+
+def rgb_PCA(rgb_image, n_components=1, get_component=None):
```

Comment:
This is a pretty cool method! Is it from a particular paper? I suggest adding a docstring and referencing the paper there.